### PR TITLE
Excess rules and meters removing

### LINKED ISSUE #2189: Add "remove excess" to /switches/{switch_id}/synchronize
Scope:
1. Add "Remove excess=(true/false)" optional parameter to `/switches/{switch_id}/synchronize`. Optional, `False` by default
2. Implement excess flow removal if flag above set to `True` as a part of synchronize operation.
System should be able to remove both excess rules and meters from the switch

### PR DESCRIPTION
**Pull request by IlyaMoiseev**
Closes #2189 
Depends on #2289

Rebased version of #2215

---
This change is [Reviewable](https://reviewable.io/reviews/telstra/open-kilda/2438)

### CODE REVIEW

File: services/src/floodlight-service/floodlight-modules/src/main/java/org/openkilda/floodlight/kafka/RecordHandler.java [hunk 1]
```diff
@@ -448,6 +456,35 @@ private void doDeleteFlow(final CommandMessage message, String replyToTopic, Des
...
+        processDeleteFlow(command, dpid);
+
+        message.setDestination(replyDestination);
+        getKafkaProducer().sendMessageAndTrack(replyToTopic, message);
+    }
+
+    /**
+     * Batch removes flow.
+     *
+     * @param message command message for flow installation
```
Comment (timofei-durakov) [inline services/src/floodlight-service/floodlight-modules/src/main/java/org/openkilda/floodlight/kafka/RecordHandler.java]: copy paste?

File: services/src/floodlight-service/floodlight-modules/src/main/java/org/openkilda/floodlight/kafka/RecordHandler.java [hunk 2]
```diff
@@ -846,6 +879,35 @@ private void doDeleteMeter(CommandMessage message) {
...
+        BatchRemoveMeters request = (BatchRemoveMeters) message.getData();
+        logger.info("Deleting meters '{}'. Switch: '{}'", request.getMetersId(), request.getSwitchId());
+
+        final IKafkaProducerService producerService = getKafkaProducer();
+        final String replyToTopic = context.getKafkaSwitchManagerTopic();
+
+        try {
+            DatapathId dpid = DatapathId.of(request.getSwitchId().toLong());
+
+            for (Long meterId : request.getMetersId()) {
```
Comment (timofei-durakov) [inline services/src/floodlight-service/floodlight-modules/src/main/java/org/openkilda/floodlight/kafka/RecordHandler.java]: is it possible to be null?

File: services/src/messaging/src/main/java/org/openkilda/messaging/info/switches/SyncRulesResponse.java [hunk 3]
```diff
@@ -21,9 +21,9 @@
 import com.fasterxml.jackson.annotation.JsonProperty;
 import lombok.Value;
 
-import java.util.Collections;
 import java.util.List;
 
+//TODO should be removed
```
Comment (timofei-durakov) [inline services/src/messaging/src/main/java/org/openkilda/messaging/info/switches/SyncRulesResponse.java]: deprecated?

File: services/src/northbound-service/northbound/src/main/java/org/openkilda/northbound/service/SwitchService.java [hunk 4]
```diff
@@ -131,6 +132,15 @@
      */
     CompletableFuture<RulesSyncResult> syncRules(SwitchId switchId);
 
+    /**
+     * Synchronize (install) missing flows that should be on the switch but exist only in Neo4J.
```
Comment (timofei-durakov) [inline services/src/northbound-service/northbound/src/main/java/org/openkilda/northbound/service/SwitchService.java]: missing rules?

File: services/src/northbound-service/northbound/src/main/java/org/openkilda/northbound/controller/v1/SwitchController.java [hunk 5]
```diff
@@ -263,6 +265,20 @@
...
     }
 
+    /**
+     * Synchronize (install) missing flows that should be on the switch but exist only in neo4j.
+     * Optionally removes excess rules from the switch.
+     *
+     * @return the synchronization result.
+     */
+    @ApiOperation(value = "Synchronize rules on the switch", response = RulesSyncResult.class)
+    @PostMapping(path = "/{switch_id}/synchronize")
```
Comment (nikitacherevko) [inline services/src/northbound-service/northbound/src/main/java/org/openkilda/northbound/controller/v1/SwitchController.java]: If we follow [API guidelines](https://github.com/telstra/open-kilda/blob/develop/docs/guidelines/api-guidelines.md) written by Sergii, then there should be a PATCH method. Please correct me if I'm wrong

File: services/src/messaging/src/main/java/org/openkilda/messaging/command/flow/BatchRemoveFlowForSwitchManagerRequest.java [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+import lombok.Value;
+
+import java.util.List;
+
+@Value
+@Builder
+public class BatchRemoveFlowForSwitchManagerRequest extends CommandData {
+
+    @JsonProperty("switch_id")
+    private SwitchId switchId;
```
[the remark this answers is not in the document]
Comment (nikitacherevko) [reply services/src/messaging/src/main/java/org/openkilda/messaging/command/flow/BatchRemoveFlowForSwitchManagerRequest.java]: Lombok will make private and final all fields by default

**Pull request merged**